src/janitor.c

```diff
@@ -400,6 +400,8 @@ static void pool_client_maint(PgPool *pool)
 				disconnect_client(client, true, "query_timeout");
 			} else if (cf_query_wait_timeout > 0 && age > cf_query_wait_timeout) {
 				disconnect_client(client, true, "query_wait_timeout");
+			} else if (cf_query_db_wait_timeout && age > cf_query_db_wait_timeout && is_pool_failing(client->pool)) {
```

Comment:
`cf_query_db_wait_timeout &&` uses a different convention than the other predicates. For example `cf_query_wait_timeout > 0 &&` using `> 0` to check if the config setting has a value.